### ISSUE TITLE
'os_sorted' sorts files with spaces in names not as in Windows Explorer

### ISSUE THREAD

**Issue by Tomarchelone** [SethMMorton/natsort#165]
**The bug**
I have two files with space in one name, say 'a.pdf' and 'a b.pdf'. `os_sorted` sorts these names as
'a.pdf'
'a b.pdf'

**Expected behavior**
Windows Explorer sorts the files in the following order:

'a b.pdf'
'a.pdf'

I expect the `os_sorted` funcion to do the same.

**Environment:**
 - Python Version: 3.11.1
 - Windows

**To Reproduce**
```
from natsort import os_sorted

print(os_sorted(['a.pdf', 'a b.pdf']))

```

If it is of any help, the code from [here](https://stackoverflow.com/questions/4813061/non-alphanumeric-list-order-from-os-listdir) seems to sort the files as in Windows Explorer. Namely:

```
from ctypes import wintypes, windll
from functools import cmp_to_key

def winsort(data):
    _StrCmpLogicalW = windll.Shlwapi.StrCmpLogicalW
    _StrCmpLogicalW.argtypes = [wintypes.LPWSTR, wintypes.LPWSTR]
    _StrCmpLogicalW.restype  = wintypes.INT

    cmp_fnc = lambda psz1, psz2: _StrCmpLogicalW(psz1, psz2)
    return sorted(data, key=cmp_to_key(cmp_fnc))


```

**Comment by SethMMorton** [comment]
This is a result of code that attempts to sort e.g. "file.txt" before "file (2).txt". This is now it sorts on my Linux machine, and if I remember correctly on Mac as well. I just checked, and Windows sorts this as "file (2).txt" before "file.txt", so this treatment is incorrect for Window. I will fix it.

**Comment by SethMMorton** [comment]
Closed with #166
